src/modules/Home/components/CategoryGrid.tsx

```diff
@@ -1,26 +1,48 @@
 import { Product } from 'types'
-import { Flex, Heading } from '@chakra-ui/react'
+import { Button, Flex, Heading} from '@chakra-ui/react'
 import ProductCard from './ProductCard'
+import {  useState } from 'react'
 
 interface Props<T> {
   title: string
   products: T[]
 }
 
 const CategoryGrid = <T extends Product>({ title, products }: Props<T>) => {
+  const[isView, setIsView] =  useState(false);
+ const newProducts: typeof products = isView ? products : products.slice(0, 12);
```

Comment:
ternar should be used in jsx. For logic description better use if/else or switch case